docs/src/account/components.md

```diff
@@ -9,180 +9,201 @@ Account components are reusable units of functionality that define a part of an
+- **Composed**: provided through `type = [ ... ]`, which contains exactly four `FeltSchema` descriptors. Each element is either a named typed field (optionally with `default-value`) or a `void` element for reserved/padding zeros.
 
-A word can be written as a hexadecimal value, and field elements can be written either as hexadecimal or decimal numbers. In all cases, numbers should be input as strings.
+Composed schema entries reuse the existing TOML structure for four-element words, while singular schemas rely on `type`. In our example, the `token_metadata` slot uses a composed schema (`type = [...]`) mixing typed fields (`max_supply`, `decimals`) with defaults (`symbol`) and a reserved/padding `void` element.
 
-In our example, the `token_metadata` single-slot entry is defined as four elements, where the first element is a placeholder, and the second, third and fourth are hardcoded values.
+##### Word schema example
+
+```toml
+[[storage.value]]
+name = "demo::faucet_id"
+description = "Account ID of the registered faucet"
+type = [
+  { type = "felt", name = "prefix", description = "Faucet ID prefix" },
+  { type = "felt", name = "suffix", description = "Faucet ID suffix" },
+  { type = "void" },
+  { type = "void" },
+]
+```
 
 ##### Word types
 
-Valid word types are `word` (default type) and `auth::rpo_falcon512::pub_key` (represents a Falcon public key). Both can be written and interpreted as hexadecimal strings.
+Singular schemas accept `word` (default) and word-shaped types such as `auth::rpo_falcon512::pub_key` or `auth::ecdsa_k256_keccak::pub_key` (parsed from hexadecimal strings).
+
+Singular schemas can also use any felt type (e.g. `u8`, `u16`, `u32`, `felt`, `token_symbol`, `void`). The value is parsed as a felt and stored as a word with the parsed felt in the last element and the remaining elements set to `0`.
 
 ##### Felt types
 
-Valid field element types are `u8`, `u16`, `u32`, `felt` (default type) and `token_symbol`:
+Valid field element types are `void`, `u8`, `u16`, `u32`, `felt` (default) and `token_symbol`:
 
-- `u8`, `u16` and `u32` values can be parsed as decimal numbers and represent 8-bit, 16-bit and 32-bit unsigned integers
-- `felt` values represent a field element, and can be parsed as decimal or hexadecimal values
-- `token_symbol` values represent the symbol for basic fungible tokens, and are parsed as strings made of four uppercase characters
+- `void` is a special type which always evaluates to `0` and does not produce an init requirement; it is intended for reserved or padding elements.
+- `u8`, `u16` and `u32` values can be parsed as decimal numbers and represent 8-bit, 16-bit and 32-bit unsigned integers.
+- `felt` values represent a field element, and can be parsed as decimal or hexadecimal numbers.
+- `token_symbol` values represent basic fungible token symbols, parsed as 1–6 uppercase ASCII characters.
 
 #### Header
```

Comment:
The flow feels a bit odd here. I would have expected that we describe the header right after we show the example, and then dive into storage entries etc.